cmd/contour/contour.go

```diff
@@ -34,10 +34,11 @@ func main() {
 	envoyCmd := app.Command("envoy", "Sub-command for envoy actions.")
 	sdm, shutdownManagerCtx := registerShutdownManager(envoyCmd, log)
 
+	bootstrap, bootstrapCtx := registerBootstrap(app)
+
 	// Add a "shutdown" command which initiates an Envoy shutdown sequence.
-	sdmShutdown, sdmShutdownCtx := registerShutdown(envoyCmd, log)
+	sdmShutdown, sdmShutdownCtx := registerShutdown(envoyCmd, bootstrapCtx.AdminPort, log)
```

Comment:
I see what you've tried to do here, but I don't think that what this is doing is what you want.

Currently, this will bring in the default port from the bootstrap context, but it won't get overridden by a parameter to the `contour shutdown`  command, so you won't actually have a way to change it.

Reply:
Hi @youngnick I hope to have correctly addressed all the requested changes now. Thanks!